### PR TITLE
Reorganize demo feature documentation

### PR DESCRIPTION
**Pull request by flands**
- Create a nested section called features
- Organize feature index page
- Add feature flags under features subsection
- Add signal coverage under features subsection

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
Content looks good overall. Could you run `npm run fix` to address formatting and stuff? Thanks!

**Comment by flands** [comment]
OK, I think I fixed the links - at least they appear to work with `npm run serve:netlify` but they don't work with npm run serve as hugo aliases appear to be disabled... Is there a different way to do this?

**Comment by cartermp** [comment]
Hmm, seems I still get these errors with `npm run check:links`:

```
blog/2023/demo-birthday/index.html
  target does not exist --- blog/2023/demo-birthday/index.html --> /docs/demo/feature-flags/
blog/2022/announcing-opentelemetry-demo-release/index.html
  target does not exist --- blog/2022/announcing-opentelemetry-demo-release/index.html --> /docs/demo/trace-features/
  target does not exist --- blog/2022/announcing-opentelemetry-demo-release/index.html --> /docs/demo/metric-features/
docs/demo/architecture/index.html
  target does not exist --- docs/demo/architecture/index.html --> /docs/demo/metric-features/
  target does not exist --- docs/demo/architecture/index.html --> /docs/demo/trace-features/
docs/demo/index.html
  target does not exist --- docs/demo/index.html --> feature-flags
  target does not exist --- docs/demo/index.html --> feature-flags/
  target does not exist --- docs/demo/index.html --> metric-features/
  target does not exist --- docs/demo/index.html --> trace-features/
  ```

**Comment by flands** [comment]
Yes, me too, and this appears to be because hugo.yaml has `disableAliases: true # We do redirects via Netlify's _redirects file`. If you want it to work, you have to run `npm run serve:netlify` instead of `npm run serve`. I'm not sure how to proceed here...

**Comment by cartermp** [comment]
Ergh, failed twice to push changes to your branch due to git permissions -- here's a patch file of the changes you need to make this work.
[0001-fix-links.patch](https://github.com/open-telemetry/opentelemetry.io/files/13798646/0001-fix-links.patch)

**Comment by puckpuck** [comment]
I like some of where this is going, but I see it adding even more confusion.

The new section includes Telemetry coverage and Feature Flags. I don't believe they shouldn't be grouped. Should the new section be called "Telemetry Features" instead, and Feature Flags be moved back up? Also, I would like to see "Manual Span Attributes" pulled into this section.

Thinking into the future, we currently have a gap with manual metrics needing to be documented, and this section would be a great spot to add them.

**Comment by flands** [comment]
> Ergh, failed twice to push changes to your branch due to git permissions -- here's a patch file of the changes you need to make this work. [0001-fix-links.patch](https://github.com/open-telemetry/opentelemetry.io/files/13798646/0001-fix-links.patch)

Thanks and yeah, I feared this was the answer - I'll file a bug about aliases not working properly. I'm not sure fixing all links is scalable long-term.

**Comment by cartermp** [comment]
@puckpuck any further thoughts on this PR?

**Review by puckpuck — APPROVED** [review]
This looks good. :shipit:

**Review by cartermp — APPROVED** [review]

File: content/en/docs/demo/feature-flags.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Feature Flags
-aliases: [feature_flags]
+aliases: [feature_flags, ../feature_flags/]
```
Comment (chalin) [inline content/en/docs/demo/feature-flags.md:3]: What was the intent here? This added alias effectively pushes a page alias outside of the Demo section, which we don't want IMHO.

Comment (chalin) [reply content/en/docs/demo/feature-flags.md:3]: Let's track this via:

- #3801

**Comment by chalin** [comment]
Btw, thanks for the PR @flands, this slight reorg is an improvement IMHO! 👍🏻

**Pull request merged**